client/lib/cart-values/cart-items.js

```diff
@@ -81,6 +82,11 @@ function cartItemShouldReplaceCart( cartItem, cart ) {
 		return true;
 	}
 
+	if ( cartItem.free_trial ) {
+		// adding a free trial plan to your cart replaces the cart
+		return true;
+	}
+
```

Comment:
We also need to consider the case when the user already has a free trial in their cart, and they add another product.